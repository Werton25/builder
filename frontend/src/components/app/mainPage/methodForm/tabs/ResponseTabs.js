import React from 'react';
import PropTypes from 'prop-types';
import Editor from "../editor/Editor";
import {Tabs, TabLink, TabContent} from 'react-tabs-redux';
import './Tabs.css';

export default class ResponseTabs extends React.Component {
  render() {
    return (
      <Tabs className='tabs tabs-1'>
        <div className='tab-links'>
          <TabLink to='response'>Ответ</TabLink>
        </div>
        <div className='content'>
          <TabContent for='response'>
            <Editor value={this.props.response}
                    options={{readOnly: true}}/>
          </TabContent>
        </div>
      </Tabs>
    );
  }
};

ResponseTabs.propTypes = {
  response: PropTypes.string
};
