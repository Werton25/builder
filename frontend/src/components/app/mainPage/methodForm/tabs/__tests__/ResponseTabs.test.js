import React from "react";
import renderer from 'react-test-renderer';
import ResponseTabs from "../ResponseTabs";

jest.mock('../../editor/Editor');
jest.mock('../../../../../../selectors/editor', () => ({isFullScreenMode: () => false}));

describe('ResponseTabs tests', () => {
  it('ResponseTabs renders correctly', () => {
    const tree = renderer
      .create(<ResponseTabs/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

