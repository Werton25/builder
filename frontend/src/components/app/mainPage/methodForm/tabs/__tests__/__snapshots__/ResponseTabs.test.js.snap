// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResponseTabs tests ResponseTabs renders correctly 1`] = `
<div
  className="tabs tabs-1"
>
  <div
    className="tab-links"
  >
    <div
      className="tab-link tab-link-active"
      onClick={[Function]}
      onKeyPress={[Function]}
      style={
        Object {
          "fontWeight": "bold",
        }
      }
      tabIndex="0"
    >
      Ответ
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="tab-content tab-content-visible"
      style={Object {}}
    />
  </div>
</div>
`;
